force-app/main/adapter/in/sobjects/contact/ContactAdapter.cls

```diff
@@ -665,22 +665,56 @@ public inherited sharing class ContactAdapter extends fflib_SObjects2 {
             Address__c newAddressFromContact = contactAddressesByContact.get(contact);
             Address__c existingAddressFromContact = existingAddressesByAddress.get(newAddressFromContact);
 
-            // if found a match
-            if (contactAddressHasAddressMatch(existingAddressFromContact)) {
+            // Check if the contact's address fields are intentionally set to null
+            Boolean isAddressFieldsNull = String.isBlank(contact.MailingStreet) &&
+                                      String.isBlank(contact.MailingCity) &&
+                                      String.isBlank(contact.MailingPostalCode) &&
+                                      String.isBlank(contact.MailingState) &&
+                                      String.isBlank(contact.MailingCountry);
+
+            // If all address fields are null, clear the Current_Address__c field and continue
+            if (isAddressFieldsNull) {
+                contact.Current_Address__c = null;
+                continue;  // Skip address creation for this contact
+            }
+            
+            // Check if the contact has Address Override enabled
+            if (contact.Is_Address_Override__c == true) {
```

Comment:
Is there a possibility that this logic will create multiple Address records for the same address over time, basically each time the Contact record is edited to change any field value? The original logic appears to have first looked for a matching address before it inserted a new address. The new changes move that check to after the logic to insert a new address.

Reply:
Thank you for bringing this up. The reason we're proceeding to create a new address when Address Override is enabled, without checking for an existing address, is to ensure that the contact’s address remains completely independent of any household address updates.
By generating a new address for each contact with Address Override enabled, we maintain distinct address records for the contact, which aligns with the intended behavior of keeping contact addresses separate from household changes. This prevents any potential conflicts between the contact’s address and the household’s address. For contacts without Address Override, we continue to match existing addresses before inserting new ones to ensure consistency.

Reply:
That makes sense. My question though is whether every tiny change to the Contact will create a new Address record. To test:
- Create a Contact with an Address and the Override set
- Confirm that there is one Address record linked to that Contact
- Edit the Contact again to add or change the value in Title
- Is another Address record created for the same exact address?

If the answer to that last bullet is yes, we'll need to fix that. Let me know what you find.

Reply:
@force2b, I have tested the scenario where non-address fields like Title are updated, and I confirmed that no new address records are created in such cases. The new address creation is only triggered by changes in address-related fields, as intended.

Reply:
Thank you for testing and confirming.